rotkehlchen/tests/unit/accounting/test_exchanges.py

```diff
@@ -38,3 +42,36 @@ def test_account_for_coinbase_ledger_actions(rotkehlchen_api_server_with_exchang
     assert events_map[AccountingEventType.FEE] == 2
     assert events_map[AccountingEventType.ASSET_MOVEMENT] == 3
     assert sum(events_map.values()) == 11  # processed events
+
+
+@pytest.mark.parametrize('default_mock_price_value', [FVal(1.5)])
+@pytest.mark.parametrize('ethereum_accounts', [[]])
+def test_exchanges_removed_api_keys(rotkehlchen_api_server):
+    """
+    Test that if actions made on an exchange are stored in the DB but the API keys were removed
+    then the actions are still taken into account in accounting.
+    """
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+    with rotki.data.db.user_write() as write_cursor:
+        rotki.data.db.add_trades(write_cursor, trades=[Trade(
+            timestamp=Timestamp(1611426201),
```

Comment:
Can you add also an asset movement? And add to the database a manual trade that doesn't belong to any exchange. location external